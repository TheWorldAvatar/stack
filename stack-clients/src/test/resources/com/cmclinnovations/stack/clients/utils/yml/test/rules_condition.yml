prefixes:
  rdfs: "http://www.w3.org/2000/01/rdf-schema#"
  base: "https://theworldavatar.io/kg/"
  grel: "http://users.ugent.be/~bjdmeest/function/grel.ttl#"

mappings:
  person:
    s: base:person/$(id)
    po:
      - p: base:hasName
        o:
          - mapping: person-name
            condition:
              function: idlab-fn:notEqual
              parameters:
                - [grel:valueParameter, $(name)]
                - [grel:valueParameter, ""]
  person-name:
    subject: base:person/name/$(id)
    po:
      - p: base:hasValue
        o: $(name)
        condition:
          function: idlab-fn:notEqual
          parameters:
            - [grel:valueParameter, $(name)]
            - [grel:valueParameter, ""]
