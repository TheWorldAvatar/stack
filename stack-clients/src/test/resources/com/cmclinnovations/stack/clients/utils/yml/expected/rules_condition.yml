prefixes:
  rdfs: "http://www.w3.org/2000/01/rdf-schema#"
  base: "https://theworldavatar.io/kg/"
  grel: "http://users.ugent.be/~bjdmeest/function/grel.ttl#"

mappings:
  person-name:
    sources: source-ref
    po:
      - p: base:hasValue
        o: $(name)
        condition:
          function: grel:boolean_not
          parameters:
            - parameter: grel:bool_b
              value:
                function: idlab-fn:isNull
                parameters:
                  - parameter: idlab-fn:str
                    value: $(name)
    s:
      value: base:person/name/$(id)
      targets: target-ref
  person:
    sources: source-ref
    s:
      value: base:person/$(id)
      targets: target-ref
    po:
      - p: base:hasName
        o:
          - mapping: person-name
            condition:
              function: grel:boolean_not
              parameters:
                - parameter: grel:bool_b
                  value:
                    function: idlab-fn:isNull
                    parameters:
                      - parameter: idlab-fn:str
                        value: $(name)
sources:
  source-ref:
    referenceFormulation: csv
    access: [source]
targets:
  target-ref:
    serialization: turtle
    type: sd
    access: [target]
